kie-remote/kie-remote-client/src/main/java/org/kie/services/client/api/command/AbstractRemoteCommandObject.java

```diff
@@ -228,15 +228,19 @@ void preprocessParameterCommand( Object cmdObj, List<Object> extraClassInstanceL
 
     void addPossiblyNullObject( Object inputObject, List<Object> objectList ) {
         if( inputObject != null ) {
-            if( inputObject instanceof List )  { 
-                objectList.addAll((List) inputObject);
+            if( inputObject instanceof List )  {
+                for( Object obj : (List<?>)inputObject ) {
+                    addPossiblyNullObject(obj, objectList);
+                }
```

Comment:
I think that that check if the object isn't already present in objectList could be tricky as client may want to send same object several times, for example as output content with different content keys.
About self-referencing issue - good point, will try to think up some solution which would cover that.